lib/galaxy/files/templates/models.py

```diff
@@ -104,17 +104,43 @@ class AzureFileSourceConfiguration(StrictModel):
     writable: bool = False
 
 
+class WebdavFileSourceTemplateConfiguration(StrictModel):
+    type: Literal["webdav"]
+    url: Union[str, TemplateExpansion]
+    root: Union[str, TemplateExpansion]
+    login: Union[str, TemplateExpansion]
+    password: Union[str, TemplateExpansion]
+    writable: Union[bool, TemplateExpansion] = False
+    use_temp_files: Optional[Union[bool, TemplateExpansion]] = None
+    temp_path: Optional[Union[str, TemplateExpansion]] = None
+    template_start: Optional[str] = None
+    template_end: Optional[str] = None
+
+
+class WebdavFileSourceConfiguration(StrictModel):
+    type: Literal["webdav"]
+    url: str
+    root: str
+    login: str
+    password: str
+    writable: bool = False
+    use_temp_files: Optional[bool] = None
```

Comment:
So in my issue tracker for these plugins (https://github.com/galaxyproject/galaxy/issues/18128) I mentioned:

> Revise timeout handling for S3FS file source plugin. There should be a global variable that is defined at the Galaxy level and applies to each instance if a value isn’t set. This feels better than having each plugin instance define a magic constant - which seems to be what main is doing unfortunately (https://github.com/galaxyproject/usegalaxy-playbook/blob/main/env/common/templates/galaxy/config/file_sources_conf.yml.j2)

I think ``use_temp_files`` is the same? Is there a reason this should be changing from plugin instance to plugin instance and not just set globally?

I am... very anxious about each new plugin and plugin attribute. I think we have some obligation to keep the interface to these things pretty stable because we're storing the blobs in the database. I think it would be better to set these variables at the Galaxy level unless we're very confident it is a detail we want to expose to users or otherwise vary from plugin to plugin.

Reply:
I agree, this particular case should always be defined as `True` globally. The user doesn't need to deal with this technical detail.

Reply:
I have removed the `use_temp_files` config parameter from the model. Though it was added to reflect the existing WebDAV setup ([EUs file source template](https://github.com/usegalaxy-eu/infrastructure-playbook/blob/master/templates/galaxy/config/file_sources_conf.yml.j2#L12-L13), [WebDAV file source code](https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/files/sources/webdav.py#L24-L33)), it was not exposed to the user via the template, and hence, it was not recorded in the `user_file_source` DB table (at least, it was not seen when listing the ` /API/file_source_instances`; hence, I assume that this was not recorded in the DB table; maybe I am wrong about this).

In the new commit, I have updated the diagrams and the `production_webdav.yml` template to include the `temp_path` config option so an admin can overwrite the default behavior (this option is not exposed to the user). Please let me know if this is fine.

Reply:
I want it out of the models all together. It should be in Galaxy's global config object and passed to the FileSource plugin by a different mechanism... I get that is a bit nebulous and more of a developer task. @davelopez do you understand what I'm asking for - is that something you have an idea about how to implement? I can prototype something if needed - it doesn't seem like it should be a huge challenge but I'm unsure.

Reply:
> do you understand what I'm asking for - is that something you have an idea about how to implement? 

@jmchilton I'm not 100% sure :sweat_smile:, but If you point me to any existing example or minimal prototype I can take a look and help @sanjaysrikakulam :+1:

Reply:
Hey @jmchilton,

I'm sorry, I was a bit occupied with other work. 

Is the following what you had in mind?

Update config schema file: https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/config/schemas/config_schema.yml

_Example:_
```
webdav_use_temp_files:
  type: bool
  default: true
  required: false
  desc: |
    <description>

webdav_temp_dir_path:
  type: str
  required: false
  desc: |
    <description>
```

update `webdav.py`: https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/files/sources/webdav.py#L19-L33

_Example:_
```
from galaxy import config # import this to access the galaxy config

class WebDavFilesSource(PyFilesystem2FilesSource):
    plugin_type = "webdav"
    required_module = WebDAVFS
    required_package = "fs.webdavfs"

    # Changes are made to this function
    def _open_fs(self, user_context=None, opts: Optional[FilesSourceOptions] = None):
        props = self._serialization_props(user_context)

        # Check if 'use_temp_files' and 'temp_path' are set in the config, otherwise fallback
        props["use_temp_files"] = getattr(config, 'webdav_use_temp_files', True)
        temp_path = getattr(config, 'webdav_temp_dir_path', None)
        
        if temp_path is None:
            # Fallback to props value or use temporary directory if not set in config
            temp_path = props.get("temp_path", tempfile.TemporaryDirectory(prefix="webdav_"))
        props["temp_path"] = temp_path

        extra_props: Union[FilesSourceProperties, dict] = opts.extra_props or {} if opts else {}
        handle = WebDAVFS(**{**props, **extra_props})
        return handle
```

We need to update the [sample config file](https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/config/sample/galaxy.yml.sample) with the new config option (`webdav_temp_dir_path`). 

If this is what you had in mind, I can update this PR accordingly and move those variables out of the template and the model. The above changes will retain the current behavior (admins setting the `use_temp_files: true`, and `temp_path: /data/tmp` in the `file_sources_conf.yml` for every instance of the `webdav` file source) and allow the admins to set the temp directory path in the config file.

Reply:
Is there a reason to ever specify the temp path on a per plugin basis ? If not just construct the parameter as needed `os.path.join(config.new_file_path, "file_sources")`. If for a reason you want to be able to set the temp path for file sources, same reasoning, make it a single variable and join with the plugin type.

Reply:
Sorry for not answering sooner - but yes - I think adding the options to config_schema is what I'd prefer. @mvdbeek's idea is likely fine also and even easier. I don't know anything about how webdav plugin uses the temp directory - I don't know if it needs to exclusive or not - but probably not right?

Reply:
Admin overload is a thing, and I can't think of a reason this should be set (if exclusive temp is required ... use tempfile.mkdtemp(dir=app.config.new_file_path, "my-plugin")). We might even be shooting ourselves in the foot if for whatever reason we have to tie the temp location to the job destination. I really think Galaxy needs to decide this, not the admin.

Reply:
The temp directory is just a place to temporarily stream big files instead of loading them fully into memory. It doesn't matter where it is set up as long as there is space for them. So it doesn't need to be set per plugin but you'd want to clean up this temp directory from time to time.

Reply:
A compromise would be a `file_source_temp_dir` setting, which defaults to new_file_path ?

Reply:
Thank you for your comments. I have added the `file_source_temp_dir` config parameter and updated the webdav file source to use it if it is available, and if not, it will fall back to the `new_file_path.` I also tried to retain the existing behavior of the webdav file source. 

Please let me know if anything further needs to be done.